File: src/clients/java/src/tigerbeetle-java/src/main/java/com/tigerbeetle/CreateTransfersResult.java
package com.tigerbeetle;

import java.nio.ByteBuffer;

public final class CreateTransfersResult {

    static final class Struct {
        public static final int SIZE = 8;
    }

    public final int index;
    public final CreateTransferResult result;

    CreateTransfersResult(ByteBuffer ptr) {
        index = ptr.getInt();
        result = CreateTransferResult.fromValue(ptr.getInt());
    }

    CreateTransfersResult(int index, CreateTransferResult result) {
        this.index = index;
        this.result = result;
    }
}

File: src/clients/dotnet/src/TigerBeetle/CreateTransfersResult.cs
using System.Runtime.InteropServices;

namespace TigerBeetle
{
    [StructLayout(LayoutKind.Sequential, Size = SIZE)]
    public struct CreateTransfersResult
    {
        #region Fields

        public const int SIZE = 8;

        private readonly int index;

        private readonly CreateTransferResult result;

        #endregion Fields

        #region Constructor

        internal CreateTransfersResult(int index, CreateTransferResult result)
        {
            this.index = index;
            this.result = result;
        }

        #endregion Constructor

        #region Properties

        public int Index => index;

        public CreateTransferResult Result => result;

        #endregion Properties
    }
}
